src/main/java/net/neoforged/neoforge/client/event/AddClientReloadListenerEvent.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright (c) Forge Development LLC and contributors
+ * SPDX-License-Identifier: LGPL-2.1-only
+ */
+
+package net.neoforged.neoforge.client.event;
+
+import net.minecraft.client.Minecraft;
+import net.minecraft.resources.ResourceLocation;
+import net.minecraft.server.packs.resources.PreparableReloadListener;
+import net.minecraft.server.packs.resources.ReloadableResourceManager;
+import net.neoforged.fml.LogicalSide;
+import net.neoforged.fml.event.IModBusEvent;
+import net.neoforged.neoforge.event.SortedReloadListenerEvent;
+import net.neoforged.neoforge.resource.VanillaClientListeners;
+import org.jetbrains.annotations.ApiStatus;
+
+/**
+ * This event allows mods to register client-side reload listeners to the resource manager.
+ * This event is fired once during the construction of the {@link Minecraft} instance.
+ * <p>
+ * This event is only fired on the {@linkplain LogicalSide#CLIENT logical client}.
+ * 
+ * @see {@link AddReloadListenerEvent} for registering server-side reload listeners.
+ */
+public class AddClientReloadListenerEvent extends SortedReloadListenerEvent implements IModBusEvent {
```

Comment:
Add is weird naming compared to Register, as it implies that things can be added multiple times somehow. Also, anyone searching for the event will probably look for "RegisterReloadListener" or similar.

Reply:
We had this discussion a bit earlier:

> As for the names, I used Add because we aren't registering anything. There is no registry for these objects, they're being added to a List<PreparableReloadListener>.

As for user discoverability, I've found that people are usually aware of the `AddReloadListenerEvent` but are _not_ aware of the `RegisterClientReloadListenersEvent`.

Reply:
Well, that's the kicker, right? A `Map` is a registry, a `List` is not :p

We have other events using `Add*` as a name, so it isn't some isolated incident here
![](https://i.imgur.com/nBVw7MW.png)